### DISCUSSION TITLE
How to get translations when rendering a custom component?

### DISCUSSION THREAD

**Discussion by vinioo** [react-querybuilder/react-querybuilder#451] [General]
Hi there! I'm using mui components and I'd like a way to access the translations or the component label by props.

here's what I have so far: 

A custom select:
``` 
const MyCustomSelect = ({ children, ...rest }) => ( // here we cant easily figure out if this props belongs to fields or combinators
  <CustomSelect select {...rest} label={children[0].key === '=' ? 'Operators' : 'Property'}> // i had to check if the first child is the "equal" operator to set the correct label
    {children}
  </CustomSelect>
); 
```

I'm replacing the mui Select with my custom select:
```  
const muiComponents = {
    Button,
    Checkbox,
    DragIndicator,
    FormControl,
    FormControlLabel,
    Input: CustomTextField as typeof Input,
    ListSubheader,
    MenuItem,
    Radio,
    RadioGroup,
    Select: MyCustomSelect as typeof Select,
    Switch,
    TextareaAutosize,
  };
```
The problem is I won't receive the label/translation props on the MyCustomSelect component, and I have to do a check to figure out if the select belongs to operators or fields, and then add the label.
I created a custom input too and in that case, the input receives the `label` prop so i didn't have to do any changes to the CustomInput.

Is there a simple solution to get the label in this case? I wanted a solution to this without having to re-create the component on the controlElements prop. 

thanks!

**Comment by jakeboone02** [comment]
Are you looking for the `title` prop? That's provided to every component. It comes from the `translations` object and is different for field selectors and operator selectors.

**Reply by vinioo** [reply]
@jakeboone02 exactly, that prop is not being provided to Select

**Reply by vinioo** [reply]
here are the props i'm receiving:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/35852229/211862032-2317c3f7-03ff-4ce1-979b-7473b120ac7d.png">

**Reply by jakeboone02** [reply]
Ah, I see what you mean. I think you need to go the other way around and wrap the `fieldSelector`/`operatorSelector` instead of the base MUI component. See this documentation with an example using Ant Design: https://react-querybuilder.js.org/docs/compat#customization. In your case, you'd `import { MaterialValueSelector, materialControlElements } from '@react-querybuilder/material'`.

**Reply by vinioo** [reply]
thanks!
